src/js/containers/LoginPage/LoginPage.js

```diff
@@ -17,19 +37,126 @@ class LoginPage extends Component {
         }
     }
 
+    openLinkInWindow(link) {
+        window.open(link, '_blank', 'toolbar=0,status=0,width=720,height=700');
+    }
+
+    handleEmailChange(email) {
+        this.setState({ email });
+    }
+
+    handleApiKeyChange(apiKey) {
+        this.setState({ apiKey });
+    }
+
+    handleLoginSubmit(e) {
+        e.preventDefault();
+        const { dispatch, config } = this.props;
+
+        let isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+        if(isHostAPILogin) {
+            //apiKey will actually be password here
+            dispatch(asyncLogin(this.state.email , this.state.apiKey));
+        } else {
+            dispatch(asyncAPILogin(this.state.email, this.state.apiKey));
+        }
+    }
+
     render() {
+        const { formatMessage } = this.props.intl;
         const { config } = this.props;
-        let isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+
+        //client login defaults
+        let title = 'component.clientLogin.form.title';
+        let inputType = 'text';
+        let inputLabel = 'component.clientLogin.form.apiKey';
+        let loginButtonText = 'component.clientLogin.form.button';
+
+        const isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+
+        if(isHostAPILogin) {
+          title = 'component.login.form.title';
+          inputType = 'password';
```

Comment:
Why did we make API Key a password input type? Most people  copy paste but some manually enter. It'll be very hard for them to write. Also since the API Key is not something most humans can memorize just by looking at it I think is will effect negatively overall.